enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,271 @@
+* Documenting some well-known metadata keys.
+
+### Non-Goals
+
+* Restricting metadata keys to those documented in this enhancement.
+    Cincinnati may continue to add and remove keys as they see fit as long as they provide at least the keys required by this enhancement.
+* Teaching Cincinnati to support requests without `channel` query parameters.
+    This might be useful for recovering from users specifying [`desiredUpdate`][api-desiredUpdate] that were not in their configured [`channel`][api-channel], but I am fine leaving recovery to the users.
+    They should have received sufficient information to adjust their channel from wherever they received the target release they are supplying.
+
+## Proposal
+
+As proposed in [this API pull request][api-pull-request], this enhancement adds a new type:
+
+```go
+// Release represents an OpenShift release in the release graph.
+// +k8s:deepcopy-gen=true
+type Release struct {
+  // version is a semantic versioning identifying the update version. When this
+  // field is part of spec, version is optional if image is specified.
+  Version string `json:"version"`
+
+  // image is a container image location that contains the update. When this
+  // field is part of spec, image is optional if version is specified and the
+  // availableUpdates field contains a matching version.
+  Image string `json:"image"`
+
+  // Metadata is metadata associated with the release, such
+  // as associated channels (under
+  // io.openshift.upgrades.graph.release.channels) or a landing page
+  // (under url).  This information is extracted from within the
+  // release image, so it is covered by the same release image
+  // signatures used to verify the image.
+  //
+  // +optional
+  Metadata map[string]string `json:"metadata,omitempty"`
+
+  // additionalMetadata is additional metadata associated with the
+  // release.  Unlike the metadata property, the additionalMetadata
+  // property contains metadata that is not covered by the release
+  // image signatures.  For example, it may contain associated
+  // channels declared by the upstream update recommendation service.
+  // This information is as reliable as the upstream and connection used
+  // to retrieve it.
+  //
+  // Consumers who require static verification or
+  // non-repudiation should avoid additionalMetadata and prefer metadata.
+  //
+  // Consumers who prefer that security but would accept the transitory
+  // verification provided by their upstream connection should prefer
+  // values in metadata and fall back to additionalMetadata if they do
+  // not find a key in metadata.
+  //
+  // Consumers who are confident in their upstream connection and who do
+  // not desire static verification or non-repudiation may wish to prefer
+  // values in additionalMetadata and fall back to metadata if they do
+  // not find a key in additionalMetadata.  In return for the reduced
```

Comment:
I am not able to wrap my head around what it means with respect to the code in CVO. How will users choose which one i.e. `Metadata` or `AdditionalMetadata` to use?

Reply:
> I am not able to wrap my head around what it means with respect to the code in CVO.

The CVO doesn't care.  It just populates the two locations from their respective sources (release image metadata -> `metadata`, Cincinnati-graph metadata -> `additionalMetadata`).

> How will users choose which one i.e. Metadata or AdditionalMetadata to use?

It depends on their requirements for non-repudiation and/or static verification vs. upstream-service corrections.  For both [user stories][1], I am recommending the consumer prefer `additionalMetadata` and fall back to `metadata`.  I can't think of a case where a consumer would prioritize non-repudiation and/or static verification, but as I linked from [the *not distinguishing based on data source* section][2], I got pushback on just having a `metadata`.  If you are pushing back against _that_ pushback, you, @abhinavdahiya, and @smarterclayton should fight it out amongst yourselves, tell me what you decide, and I'll update the enhancement to reflect that decision.

[1]: https://github.com/openshift/enhancements/blob/c62a48bcea194bef47dbdbe749dabc25be591604/enhancements/update/available-update-metadata.md#user-stories
[2]: https://github.com/openshift/enhancements/blob/c62a48bcea194bef47dbdbe749dabc25be591604/enhancements/update/available-update-metadata.md#not-distinguishing-based-on-data-source

Reply:
Thanks for explaining. The separation between metadata and UpstreamMetadata looks fine to me.